### DISCUSSION TITLE
Create Custom Alerts

### DISCUSSION THREAD

**Discussion by cmchauvin** [teslamate-org/teslamate#1053] [Ideas]
**What would you like to be added**: Ability to setup alerts. For example, when the odometer hits a certain mileage - or x miles have been driven, alert me that it's time to rotate tires. When internal temperature hits a certain point, notify me.  

**Why is this needed**: Currently the Tesla app doesn't support any IFTTT type functions.

**Additional context**:

**Comment by pmboothby** [comment]
This is quite easy to do with Node-RED. If you're running Teslamate on Docker, you can just add a Node-red container and send the alerts to your email, etc. I've been thinking about writing a Guide for the Teslamate doco on how to set up Node-red with examples of alerts. If you're interested, perhaps I'll do it sooner than later. :)

**Reply by BriCo84** [reply]
Did you ever get around to writing a guide?  I would love to be able to do this but don't know anything about node-red

**Reply by pmboothby** [reply]
No, I didn't. I'll have a go at it this weekend. I've  just updated my docker server so some of this is fresh in my mind. I'll include some example flows:

- charging dashboard
- integrate with Telegram for messaging (easy to substitute with something else like email by changing one node)
  - send notifications with state changes
  - send periodic notifications while car is charging
  - send notifications when car goes into/out of Geofences

**Comment by cmchauvin** [comment]
Would be much appreciated. I was successful setting up Node-RED but for the life of me I can't figure out MQTT (far from experienced on this one). I'm not using my own broker - when I navigate to my virtual host address and port 1883 nothing is returned. Within Node-RED the MQTT server is just always in a "connecting" status.

**Reply by pmboothby** [reply]
Here's a draft of my Teslamate/Node-RED integration guide:

https://github.com/pmboothby/teslamate/blob/master/website/docs/integrations/Node-RED.md

I've submitted a pull request to have it added to the Teslamate documentation site, but if anyone wants to try this out and give me feedback on the documentation, that would be handy.

Though not included in this guide, there is a Tesla Node-RED module that can send commands to the car, so you could program logic to perform some event like turn on the climate control based on a calendar or timer, etc.

**Reply by henryhm0** [reply]
thanks bob, how can I create an alert using Grafana if no data is getting pull from the car? Just I found out Teslamate was not working for the last week as a result of token errors. it would be nice if I receive an alert to google hangout if that happen again or an email.
